pkg/network/go/bininspect/symbols.go

```diff
@@ -153,6 +156,9 @@ func getSymbolsUnified(f *elf.File, typ elf.SectionType, filter symbolFilter, is
 		// symbol index. If the current symbol index is a multiplier of symbolChunkSize, then we read a new chunk.
 		symbolIndex := (readLocation / symbolSizeUint64) - 1
 		if symbolIndex%symbolChunkSize == 0 {
+			if symbolSection.ReaderAt == nil {
+				return nil, fmt.Errorf("symbol section not available in random-access form")
+			}
```

Comment:
that shouldn't be inside the for-loop, but in the beginning of the function
we cannot succeed without a reader, so bail out early as possible

Reply:
Is there a guarantee this branch of the loop will always be taken? It wasn't obvious from the code.

Reply:
yes, unless `symbolSection.Size` is less than the size of a single symbol, or we failed in another check prior to the loop